sw/device/lib/dif/dif_alert_handler.c

```diff
@@ -50,21 +49,28 @@ static bool classify_alerts(const dif_alert_handler_t *handler,
     return false;
   }
 
-  uint32_t enable_reg = mmio_region_read32(
-      handler->params.base_addr, ALERT_HANDLER_ALERT_EN_SHADOWED_0_REG_OFFSET);
-  uint32_t alerts_reg =
-      mmio_region_read32(handler->params.base_addr,
-                         ALERT_HANDLER_ALERT_CLASS_SHADOWED_0_REG_OFFSET);
-
   for (int i = 0; i < class->alerts_len; ++i) {
     if (class->alerts[i] >= handler->params.alert_count) {
       return false;
     }
 
-    // Note: the value in alerts[i] corresponds directly to the bit index within
-    // the register. (I.e., alert N has enable bit N).
-    enable_reg = bitfield_bit32_write(enable_reg, class->alerts[i], true);
-
+    // Enable the alert.
+    // NOTE: the value in alerts[i] corresponds directly to the multireg index.
+    // (I.e., alert N has enable multireg N).
+    ptrdiff_t enable_reg_offset = ALERT_HANDLER_ALERT_EN_SHADOWED_0_REG_OFFSET +
+                                  (class->alerts[i] * sizeof(uint32_t));
```

Comment:
Technically you don't need the parentheses... but you can leave them if you like.

Reply:
Fixed.